moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -1060,13 +1063,41 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose& ik_pose, c
     ROS_DEBUG_STREAM_NAMED("ikfast", "No IK solution");
   }
 
+  // Among the solutions under limits, find the one that is closest to ik_seed_state
+  if (solutions_obey_limits.size() != 0)
+  {
+    double mindist = DBL_MAX;
+    int minindex = -1;
+    std::vector<double> sol;
+
+    for (size_t i = 0; i < solutions_obey_limits.size(); ++i)
+    {
+      sol = solutions_obey_limits[i];
+      double dist = harmonize(ik_seed_state, sol);
+      ROS_INFO_STREAM_NAMED("ikfast", "Dist " << i << " dist " << dist);
+
+      if (minindex == -1 || dist < mindist)
+      {
+        minindex = i;
+        mindist = dist;
+      }
+    }
+    if (minindex >= 0)
+    {
+      solution = solutions_obey_limits[minindex];
+      harmonize(ik_seed_state, solution);
+    }
+    error_code.val = moveit_msgs::MoveItErrorCodes::SUCCESS;
+    return true;
+  }
+
```

Comment:
I'd approach this by sorting the solutions in ascending order using [sort](http://www.cplusplus.com/reference/algorithm/sort/) and returning the first element from the sorted solutions vector.  This could be useful in the `getPositionIK` that returns multiple solutions.  This is only a suggestion, I'm OK with your method too.